Userland/Libraries/LibGfx/ImageFormats/CCITTDecoder.cpp

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Array.h>
+#include <AK/BitStream.h>
+#include <AK/MemoryStream.h>
+#include <LibGfx/Color.h>
+#include <LibGfx/ImageFormats/CCITTDecoder.h>
+
+namespace Gfx::CCITT {
+
+namespace {
+
+struct Code {
+    u16 run_length {};
+    u8 code_length {};
+    u16 code {};
+};
+
+constexpr Array white_terminating_codes = {
```

Comment:
Can we get some spec comments here? Official spec is probably one of these: https://www.itu.int/rec/T-REC-T/en

https://www.fileformat.info/mirror/egff/ch09_05.htm is a good overview and probably also worth linking to.